### PR TITLE
Adds Git SSH Key Generation

### PR DESCRIPTION
**Pull request by ChrisJBurns**
Signed-off-by: ChrisJBurns <29541485+ChrisJBurns@users.noreply.github.com>

#### What this PR does / why we need it:
Adds automatic Git SSH key generation if configured by user of Chart.

#### Which issue this PR fixes
https://github.com/fluxcd-community/helm-charts/issues/101

#### Checklist

- [x] [DCO](https://github.com/fluxcd-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] helm-docs are updated
- [x] Helm chart is tested
- [x] Update artifacthub.io/changes in Chart.yaml
- [ ] Run `make reviewable`

### CODE REVIEW

**Comment by ChrisJBurns** [comment]
make reviewable

**Comment by ChrisJBurns** [comment]
Getting errors when running `make reviewable`

```log
20:04:24 [ .. ] Update chartname and chartversion string in test snapshots for flux2.
sed: illegal option -- s
usage: sed script [-Ealnru] [-i extension] [file ...]
	sed [-Ealnu] [-i extension] [-e script] ... [-f script_file] ... [file ...]
sed: illegal option -- s
usage: sed script [-Ealnru] [-i extension] [file ...]
	sed [-Ealnu] [-i extension] [-e script] ... [-f script_file] ... [file ...]
```

File: charts/flux2-sync/templates/generate-git-secret.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      serviceAccountName: "{{ .Release.Name }}-generate-git-secret"
+      containers:
+      - name: flux-cli
+        image: {{ template "template.image" .Values.cli }}
+        command: 
+          ["/usr/local/bin/flux", "create",  "secret", "git", {{ .Release.Name }},
+          "--url={{ .Values.gitRepository.spec.url }}",
+          "--ssh-key-algorithm={{ .Values.secret.generate.sshKeyAlgorithm }}",
+          "--ssh-ecdsa-curve={{ .Values.secret.generate.sshEcdsaCurve }}",
+          "--namespace", {{ .Release.Namespace }}
```
Comment (stefanprodan) [inline charts/flux2-sync/templates/generate-git-secret.yaml:38]:
```suggestion
          "--namespace={{ .Release.Namespace }}",
```

**Comment by haarchri** [comment]
@ChrisJBurns can you Check your DCO Signing ?

**Comment by ChrisJBurns** [comment]
@haarchri  Ahh it must not have been signed when I clicked auto commit on Stefans change. I will do a rebase and try and fix why the tests are failing as well, not sure why they're failing at the moment. I also need to figure out why the `make reviewable` step is failing too when I run it locally.

**Comment by haarchri** [comment]
Think you need to Update the SnapshotFiles because of Chart bump

**Review by haarchri — APPROVED** [review]
tested on our environment LGTM

**Pull request merged**